### PR TITLE
Add "Cargo: New" command

### PR DESCRIPTION
**Pull request by Gordon-F**
Add support for "Cargo: New" command.

Usage:
1. Open workspace folder
2. Execute cargo new command (project or library)

Issues #169

### CODE REVIEW

File: src/services/commandService.ts [hunk 1]
```diff
@@ -404,6 +410,35 @@ export class CommandService {
...
+        }).then((name: string) => {
+            if (!name || name.length === 0) {
+                vscode.window.showErrorMessage('Invalid name');
+                return;
+            }
+            let args = ['new', name];
+            if (!isLib) {
+                args.concat('--bin');
+            }
+            let cwd = vscode.workspace.rootPath;
```
Comment (KalitaAlexey) [inline src/services/commandService.ts]: I think you should check for whether it a workspace before taking input from an user.

File: src/services/commandService.ts [hunk 2]
```diff
@@ -404,6 +410,35 @@ export class CommandService {
         return true;
     }
 
+    private static runCargoNew(isLib = false): void {
```
Comment (KalitaAlexey) [inline src/services/commandService.ts]: I do not like default parameter value. It is used when the default parameter value is what we need in most of cases. Here we have only two type of cases.

File: src/services/commandService.ts [hunk 3]
```diff
@@ -404,6 +410,35 @@ export class CommandService {
...
+        vscode.window.showInputBox({
+            placeHolder: 'Enter '.concat(isLib ? 'library' : 'project').concat(' name')
```
Comment (KalitaAlexey) [inline src/services/commandService.ts]: How about to change this to "library project" and "executable project?

File: src/services/commandService.ts [hunk 4]
```diff
@@ -172,6 +172,12 @@ export class CommandService {
     private static currentTask: CargoTask;
     public static errorFormat: ErrorFormat;
 
+    public static cargoCommandNew(commandName: string, isLib = false): vscode.Disposable {
```
Comment (KalitaAlexey) [inline src/services/commandService.ts]: The default parameter value for `isLib` is not used.

**Comment by Gordon-F** [comment]
@KalitaAlexey Thanks for review!

**Comment by KalitaAlexey** [comment]
Hi @Gordon-F,

I haven't actually tested your solution.
I have some doubts.
I am going to check it later.

Regards,
Alexey

**Comment by KalitaAlexey** [comment]
Hi @Gordon-F,

I have added your changes to my repository: https://github.com/KalitaAlexey/vscode-rust

Regards,
Alexey

**Comment by Gordon-F** [comment]
@KalitaAlexey Thanks!  How about move your fork to organization ? And find second maintainer.

**Comment by KalitaAlexey** [comment]
@Gordon-F,
I have thought about it. Can we have a chat in gitter chat instead of chatting here?
https://gitter.im/vscode-rust/Lobby

**Pull request closed without merging**